src/printer.js

```diff
@@ -98,8 +98,7 @@ function genericPrint(path, options, printPath, args) {
 
       if (
         node.decorators.length === 1 &&
-        (decorator.type === "Identifier" ||
```

Comment:
I put it back in but made an exception for classes:

```ts
@Decorarte
class Foo {}
```

There will always be a hardline after a decorator if it decorates a class. That behaviour is consistent with how exported class declarations are handled.